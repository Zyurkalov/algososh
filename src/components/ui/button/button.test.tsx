import { render, screen } from '@testing-library/react';
import renderer from "react-test-renderer";
import userEvent from '@testing-library/user-event';
import { Button } from './button';

describe("Button ui-component", () => {

    it('рендерится c текстом', () => {
        const buttonText = "Click me"

        const tree = renderer
            .create(<Button text = {buttonText} />)
            .toJSON();
            expect(tree).toMatchSnapshot();
    }); 
    
    it('рендерится без текста', () => {
        const tree = renderer
            .create(<Button />)
            .toJSON();
            expect(tree).toMatchSnapshot();
    }); 

    it('вызывает колбек функцию при клике', () => {
        const handleClick = jest.fn();
        render(<Button data-testid = 'buttonTest' onClick={handleClick}/>)

        const buttonElement = screen.getByTestId('buttonTest');
        userEvent.click(buttonElement)
        expect(handleClick).toBeCalledTimes(1);
    }); 

    it('заблокирована и не активна', () => {
        const handleClick = jest.fn();
        // render(<Button data-testid = 'buttonTest' onClick={handleClick} disabled={true}/>)

        // const buttonElement = screen.getByTestId('buttonTest');
        // userEvent.click(buttonElement)

        // expect(handleClick).not.toBeCalled();
        // expect(buttonElement).toHaveAttribute('disabled');

        const tree = renderer
            .create(<Button onClick={handleClick} disabled={true}/>)
            .toJSON();
            expect(tree).toMatchSnapshot();
            expect(handleClick).not.toBeCalled();
    }); 

    it('заблокирована индикатором загрузки и не активна', () => {
        const handleClick = jest.fn();
        // render(<Button data-testid = 'buttonTest' onClick={handleClick} isLoader={true}/>)

        // const buttonElement = screen.getByTestId('buttonTest');
        // userEvent.click(buttonElement)

        // expect(handleClick).not.toBeCalled();
        // expect(buttonElement).toHaveAttribute('disabled');

        const tree = renderer
        .create(<Button data-testid = 'buttonTest' onClick={handleClick} isLoader={true}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
        expect(handleClick).not.toBeCalled();
    });
})
