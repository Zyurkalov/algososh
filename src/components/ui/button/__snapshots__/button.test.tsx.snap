// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button ui-component заблокирована и не активна 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  onClick={[MockFunction]}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button ui-component заблокирована индикатором загрузки и не активна 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  data-testid="buttonTest"
  disabled={true}
  onClick={[MockFunction]}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button ui-component рендерится c текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Click me
  </p>
</button>
`;

exports[`Button ui-component рендерится без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
